File: CodingInterviews/LeetCode/DeleteNode.java
package LeetCode;

class Solution {
    public ListNode deleteNode(ListNode head, int val) {
        if (head == null)
            return head;
        if (head.val == val)
            return head.next;
        ListNode p = head.next, q = head;
        while (p != null) {
            if (p.val == val)
                q.next = p.next;
            p = p.next;
            q = q.next;
        }
        return head;
    }
}

File: LeetCode/Explore/Tencent/LinkList/DeleteNode.cpp
#include "../../main.h"

class Solution {
  public:
    void deleteNode(ListNode *node) {
        ListNode *p = node;
        while (node->next != nullptr) {
            node->val = node->next->val;
            p = node;
            node = node->next;
        }
        p->next = nullptr;
        delete node;
    }
};